File: dat_learning_system-fe/src/types/course.ts
import type { Lesson } from "./classroom";
import type { Topic } from "./topic";

export interface Course {
    id: string; // Guid from .NET
    category: string;
    title: string;
    isMandatory: boolean;
    thumbnail: string;
    description: string;
    certificationImage?: string;
    totalHours: number;
    rating: number;
    badge: 'Beginner' | 'Intermediate' | 'Advanced' | string;
    enrolledCount: number;
    status: 'Published' | 'Draft' | 'Closed' | string;
    topics?: Topic[];
    lessons?: Lesson[];
}

// This helps React track the form before it's converted to FormData
export interface CreateCourseInputs extends Omit<Partial<Course>, 'thumbnail' | 'title' | 'description'> {
    title: string;          // Forced required for the form
    description: string;    // Forced required for the form
    badge: string;          // Forced required for the form
    thumbnailFile?: File | null;
}

File: dat_learning_system-be/LMS.Backend/Data/Entities/Course.cs
using LMS.Backend.Common;

namespace LMS.Backend.Data.Entities;

public class Course
{
    public Guid Id { get; set; }
    
    // Changed from Enum to String to allow "Custom" creator input
    public string Category { get; set; } = "General"; 
    
    public string Title { get; set; } = string.Empty;
    public bool IsMandatory { get; set; }
    public string Thumbnail { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string CertificationImage { get; set; } = string.Empty;
    public double TotalHours { get; set; }
    public double Rating { get; set; }
    public CourseBadge Badge { get; set; }
    public int EnrolledCount { get; set; }
    
    // Soft delete logic: Use 'Closed' to hide courses without deleting them from DB
    public CourseStatus Status { get; set; } = CourseStatus.Draft;

    public List<Topic> Topics { get; set; } = new();
    public List<Lesson> Lessons { get; set; } = new();
}